server/pulp/server/db/model/__init__.py

```diff
@@ -359,7 +359,10 @@ def _write_pem_file(self, config_key, path):
                 misc.mkdir(os.path.dirname(self._pki_path))
                 os.mkdir(self._pki_path, 0700)
             with os.fdopen(os.open(path, os.O_WRONLY | os.O_CREAT, 0600), 'w') as pem_file:
-                pem_file.write(self.config[config_key])
+                try:
+                    pem_file.write(self.config[config_key])
+                except UnicodeEncodeError:
+                    pem_file.write(self.config[config_key].encode('ascii', 'replace'))
```

Comment:
Same as in pulp/nectar#63, I think you need `self.config[config_key].encode('utf-8')`.

Reply:
And I guess no `try/except` block is needed, you can always encode the string into UTF-8.